core/src/main/scala/kafka/log/Log.scala

```diff
@@ -209,6 +209,11 @@ case object LeaderOffsetIncremented extends LogStartOffsetIncrementReason {
 case object SegmentDeletion extends LogStartOffsetIncrementReason {
   override def toString: String = "segment deletion"
 }
+
+case object SegmentCompaction extends LogStartOffsetIncrementReason {
```

Comment:
It seems a little more intuitive if we swap the two new names:

```scala
case object LogCompaction extends LogStartOffsetIncrementReason
case object SegmentCompaction extends SegmentDeletionReason
```

Reply:
I took a quick look at the others that extend LogStartOffsetIncrementReason and SegementDeletionReason. I think this is why I chose Log and Segment respectively.
```
RetentionMsBreach extends SegmentDeletionReason 
RetentionSizeBreach extends SegmentDeletionReason 
StartOffsetBreach extends SegmentDeletionReason
LogRecovery extends SegmentDeletionReason
LogTruncation extends SegmentDeletionReason 
LogRoll extends SegmentDeletionReason 
LogDeletion extends SegmentDeletionReason 
```

```
case object ClientRecordDeletion extends LogStartOffsetIncrementReason
LeaderOffsetIncremented extends LogStartOffsetIncrementReason 
SegmentDeletion extends LogStartOffsetIncrementReason 
SnapshotGenerated extends LogStartOffsetIncrementReason 
```

Reply:
But I think it also makes sense to switch them.